src/overlays/actors/ovl_En_Gamelupy/z_en_gamelupy.c

```diff
@@ -32,37 +35,161 @@ const ActorInit En_Gamelupy_InitVars = {
+static Color_RGBA8 sPrimColor = { 255, 255, 255, 255 };
+
+static Color_RGBA8 sEnvColor = { 100, 200, 0, 255 };
+
+void EnGamelupy_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnGamelupy* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.03f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 10.0f);
+    this->actor.shape.yOffset = 700.0f;
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.gravity = -0.5f;
+    this->actor.shape.rot.y = Rand_Next();
+    this->unk_19C = 0;
+    this->unk_19E = 0;
+    this->unk_1A0 = 0x7D0;
+    if (this->actor.params == 1) {
+        this->rupeeIndex = 1;
+    } else {
+        this->rupeeIndex = 0;
+    }
+    func_80AF6944(this);
+}
+
+void EnGamelupy_Destroy(Actor* thisx, PlayState* play) {
+    EnGamelupy* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void func_80AF6854(EnGamelupy* this, PlayState* play) {
+    Vec3f sp4C;
+    Vec3f sp40;
+    Vec3f sp30;
+
+    sp4C = this->actor.world.pos;
+    sp30.x = Math_SinS(this->unk_19E) * 3.0f;
+    sp30.y = 5.5f;
+    sp30.z = Math_CosS(this->unk_19E) * 3.0f;
+    sp40.x = -0.05f * sp30.x;
+    sp40.y = -0.4f;
+    sp40.z = -0.05f * sp30.z;
+    EffectSsKirakira_SpawnDispersed(play, &sp4C, &sp30, &sp40, &sPrimColor, &sEnvColor, 3000, 40);
+    this->unk_19E += this->unk_1A0;
+}
+
+void func_80AF6944(EnGamelupy* this) {
+    this->actionFunc = func_80AF6958;
+}
+
+void func_80AF6958(EnGamelupy* this, PlayState* play) {
+    void* actor = func_800B6680(play, ACTOR_EN_GAMELUPY);
+
+    if (actor != NULL) {
+        this->unk_198 = actor;
+        func_80AF6994(this);
+    }
+}
+
+void func_80AF6994(EnGamelupy* this) {
+    this->actionFunc = func_80AF69A8;
+}
+
+void func_80AF69A8(EnGamelupy* this, PlayState* play) {
+    s16* temp_v0;
+
+    if ((this->collider.base.ocFlags1 & OC1_HIT)) {
+        temp_v0 = this->unk_198;
+        *temp_v0 += 0x32;
+        if (this->rupeeIndex == 1) {
+            func_801159EC(5);
+        } else {
+            func_801159EC(1);
+        }
+        func_80AF6A38(this);
+    }
+    this->actor.shape.rot.y += 0x1F4;
+}
+
+void func_80AF6A38(EnGamelupy* this) {
+    this->unk_19C = 0;
+    this->actor.gravity = 0.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_GET_RUPY);
+    this->actionFunc = func_80AF6A78;
+}
+
+void func_80AF6A78(EnGamelupy* this, PlayState* play) {
+    f32 scale;
+    Player* player = GET_PLAYER(play);
+
+    if (this->unk_19C >= 0x1F) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->unk_19C++;
+        this->actor.world.pos = player->actor.world.pos;
+        this->actor.world.pos.y += 40.0f;
+        scale = (30.0f - this->unk_19C) * 0.001f;
+        Actor_SetScale(&this->actor, scale);
+        func_80AF6854(this, play);
+    }
+    this->actor.shape.rot.y += 1000;
```

Comment:
```suggestion
    this->actor.shape.rot.y += 0x3E8;
```
Rotation so should be in hex for now.